pkg/server/render.go

```diff
@@ -120,10 +120,39 @@ func (ctrl *Controller) renderDiffHandler(w http.ResponseWriter, r *http.Request
 
 	leftOut.Tree, rghtOut.Tree = tree.CombineTree(leftOut.Tree, rghtOut.Tree)
 	fs := tree.CombineToFlamebearerStruct(leftOut.Tree, rghtOut.Tree, p.maxNodes)
-	res := renderResponse(fs, out)
+
+	var merged mergedFlamebearer
+	merged = mergedFlamebearer{
+		fs,
+		int(leftOut.Tree.Samples()),
+		int(rghtOut.Tree.Samples()),
+	}
+
+	// TODO
+	merged.SpyName = out.SpyName
+	merged.SampleRate = out.SampleRate
+	merged.Units = out.Units
+	res := map[string]interface{}{
+		"timeline":    out.Timeline,
+		"flamebearer": merged,
+		"metadata": map[string]interface{}{
+			"format":     fs.Format, // "single" | "double"
+			"spyName":    out.SpyName,
+			"sampleRate": out.SampleRate,
+			"units":      out.Units,
+		},
+	}
+
+	//	res := renderResponse(merged, out)
```

Comment:
Oh, we definitely have some tech dept here :) In fact `Flamebearer` type should be refactored to keep only related info:

```go
type Flamebearer struct {
	Names    []string `json:"names"`
	Levels   [][]int  `json:"levels"`
	NumTicks int      `json:"numTicks"`
	MaxSelf  int      `json:"maxSelf"`
	// TODO: see note in render.go
	// These fields are not related to flamebearer and should be removed.
	// Instead, the fields should be put to the response struct.
	SpyName    string `json:"spyName"`
	SampleRate uint32 `json:"sampleRate"`
	Units      string `json:"units"`
	Format     Format `json:"format"`
}
```

Thus, I wouldn't "wrap" `Flamebearer` type with two more fields but put them at the response struct root level.